### PR TITLE
Feat: Default orch config, broader detector options

### PR DESCRIPTION
**Pull request by RobGeada**
Adds:
1) A default orchestrator configuration option, which removes some boilerplate from the config
2) Adds an optional `server` field to the detector config, which defaults to be equal to `name` if not set. This lets you reuse the same detector server with different parameters in the route config, e.g.,:

```yaml
    detectors:
      - name: language-check
        server: built-in-detector
        input: true
        output: false
        detector_params:
          regex:   
            - \b(?i:darn|shoot|crud)\b
      - name: fruit-check
        server: built-in-detector
        input: true
        output: false
        detector_params:
          regex:    
            - \b(?i:orange|apple|cranberry|pineapple|grape)\b
    routes:
      - name: fruit
        detectors:
          - fruit-check
      - name: swears
        detectors:
          - language-check
      - name: passthrough
        detectors:
```

The caveat is that no route may use multiple input or output detectors that use the same server- this is not permitted by the orchestrator API.

## Summary by Sourcery

Add defaults and validation for orchestrator and detector configuration to simplify setup and enforce unique server usage

New Features:
- Provide a default OrchestratorConfig so users can omit it in the YAML
- Allow detectors to specify an optional server field that defaults to the detector name

Enhancements:
- Enforce that each route cannot have multiple input or output detectors sharing the same server
- Use detector.server as the key when building orchestrator input/output maps instead of detector name

Tests:
- Add tests for defaulting the server field
- Add tests to verify validation panics on duplicate input or output servers

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -52,6 +76,40 @@ pub fn validate_registered_detectors(gateway_cfg: &GatewayConfig) {
...
+        }
+        let mut seen = HashSet::new();
+        for server in &output_servers {
+            if !seen.insert(server) {
+                panic!(
+                    "route '{}' contains more than one output detector with server '{}'",
+                    route.name, server
+                );
+            }
+        }
```
Comment (adolfo-ab) [inline src/config.rs:112]: I think this can be done in a single pass:
```suggestion
        let mut seen_input = HashSet::new();
        let mut seen_output = HashSet::new();

        for detector_name in &route.detectors {
            if let Some(detector_cfg) = gateway_cfg.detectors.iter().find(|d| &d.name == detector_name) {
                if detector_cfg.input {
                    let server = detector_cfg.server.as_ref().unwrap();
                    if !seen_input.insert(server) {
                        panic!("route '{}' contains more than one input detector with server '{}'", route.name, server);
                    }
                }
                if detector_cfg.output {
                    let server = detector_cfg.server.as_ref().unwrap();
                    if !seen_output.insert(server) {
                        panic!("route '{}' contains more than one output detector with server '{}'", route.name, server);
                    }
                }
            }
        }
```

Comment (sheltoncyril) [reply src/config.rs:112]: For better error reporting, we could also collect all these and panic after this, raising issues with multiple detectors and not just the first one found to be duplicated.

Comment (sheltoncyril) [inline src/config.rs:112]: This might work too
```suggestion
        let mut input_servers = Vec::new();
        let mut output_servers = Vec::new();
        let mut seen_input = HashSet::new();
        let mut seen_output = HashSet::new();
	let errors = Vec::new();
        for detector_name in &route.detectors {
            if let Some(detector_cfg) = gateway_cfg.detectors.iter().find(|d| &d.name == detector_name) {
                if detector_cfg.input {
                    input_servers.push(detector_cfg.server.as_ref().unwrap());
                    if !seen_input.insert(server) {
                	errors.push(
                    	format!(“route '{}' contains more than one input detector with server '{}'", route.name, server);
            	    }
                }
                if detector_cfg.output {
                    output_servers.push(detector_cfg.server.as_ref().unwrap());
                    if !seen_output.insert(server) {
                	errors.push(
                    	format!(“route '{}' contains more than one input detector with server '{}'", route.name, server);
            	    }
                }
            }
        }
        // if errors then panic
	if errors.len() > 0 {
	panic!(errors); // Not sure whats the best way to raise it but you get the point
}
```

**Review by sheltoncyril — APPROVED** [review]
/lgtm
